web/pdf_attachment_viewer.js

```diff
@@ -89,6 +89,34 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
     /**
      * @private
      */
+    _bindPdfLink:
+        function PDFAttachmentViewer_bindPdfLink(button, content, filename) {
+      var blobUrl;
+      button.onclick = function() {
+        if (!blobUrl) {
+          blobUrl = pdfjsLib.createObjectURL(content, 'application/pdf') +
+            '#' + filename;
+        }
+        var viewerUrl;
+        if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
+          // The current URL is the viewer, let's use it and append the file.
+          viewerUrl = '?file=' + encodeURIComponent(blobUrl);
+        } else if (PDFJSDev.test('CHROME')) {
+          // eslint-disable-next-line no-undef
+          viewerUrl = chrome.runtime.getURL('/content/web/viewer.html') +
+            '?file=' + encodeURIComponent(blobUrl);
+        } else {
+          // Let Firefox's content handler catch the URL and display the PDF.
+          viewerUrl = blobUrl;
```

Comment:
Oh I see. It turns out that there are two ways to obtain the file name.

- `getFilenameFromUrl` (originally introduced in 83b6eaed1602ec3ddcbfe6633b943ecfb3da3e10)
- `getPDFFileNameFromURL` (84ae29c21d0bcdad09cb8ff71d53d4a149100fac from #3161)

I have updated the PR to use the latter.